src/pages/posts/[slug]/index.astro

```diff
@@ -22,9 +22,11 @@ const { Content } = await post.render()
 
 const { title, description } = post.data
 const date = longDate(post.data.date)
+
+const ogImage = new URL(`/posts/${post.slug}.png`, Astro.url.origin).href
```

Comment:
I need to change this to reference the site base url so absolute paths get generated in production.